File: src/angular/manager/src/app/portfolios/service/quote.service.ts
/**
 *    Copyright 2019 Sven Loesekann
   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at
       http://www.apache.org/licenses/LICENSE-2.0
   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Quote } from '../model/quote';

export enum ComparisonIndex {
	SP500 = 'IVV',
	EUROSTOXX50 = 'SXRT.DE',
	MSCI_CHINA = 'ICGA.DE'
}

@Injectable()
export class QuoteService {
  constructor(private http: HttpClient) { }

  getAllDailyQuotes(symbol: string): Observable<Quote[]> {
	return this.http.get<Quote[]>(`/rest/quote/daily/all/symbol/${symbol}`);
  }

  getIntraDayQuotes(symbol: string): Observable<Quote[]> {
	return this.http.get<Quote[]>(`/rest/quote/intraday/symbol/${symbol}`);
  }

  getDailyQuotesFromStartToEnd(symbol: string, start: Date, end: Date): Observable<Quote[]> {
	const startStr = start.toISOString().split('T')[0];
	const endStr = end.toISOString().split('T')[0];
	return this.http.get<Quote[]>(`/rest/quote/daily/symbol/${symbol}/start/${startStr}/end/${endStr}`);
  }

  getDailyQuotesForComparisonIndexFromStartToEnd(portfolioId: number, comparisonIndex: ComparisonIndex, start: Date, end: Date): Observable<Quote[]> {
	const startStr = start.toISOString().split('T')[0];
	const endStr = end.toISOString().split('T')[0];
	return this.http.get<Quote[]>(`/rest/quote/daily/portfolio/${portfolioId}/index/${comparisonIndex}/start/${startStr}/end/${endStr}`);
  }

  getDailyQuotesForComparisonIndex(portfolioId: number, comparisonIndex: ComparisonIndex): Observable<Quote[]> {
	return this.http.get<Quote[]>(`/rest/quote/daily/all/portfolio/${portfolioId}/index/${comparisonIndex}`);
  }  
}

File: src/main/java/ch/xxx/manager/usecase/service/QuoteService.java
/**
 *    Copyright 2019 Sven Loesekann
   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at
       http://www.apache.org/licenses/LICENSE-2.0
   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
 */
package ch.xxx.manager.usecase.service;

import java.time.LocalDate;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import ch.xxx.manager.domain.model.dto.QuoteDto;
import ch.xxx.manager.domain.model.entity.DailyQuoteRepository;
import ch.xxx.manager.domain.model.entity.IntraDayQuoteRepository;
import ch.xxx.manager.usecase.mapping.QuoteMapper;

@Service
@Transactional
public class QuoteService {
	private final DailyQuoteRepository dailyQuoteRepository;
	private final IntraDayQuoteRepository intraDayQuoteRepository;
	private final QuoteMapper quoteMapper;

	public QuoteService(DailyQuoteRepository dailyQuoteRepository, IntraDayQuoteRepository intraDayQuoteRepository,
			QuoteMapper quoteMapper) {
		this.dailyQuoteRepository = dailyQuoteRepository;
		this.intraDayQuoteRepository = intraDayQuoteRepository;
		this.quoteMapper = quoteMapper;
	}

	public List<QuoteDto> getDailyQuotes(String symbol) {
		return this.dailyQuoteRepository.findBySymbol(symbol).stream()
				.flatMap(quote -> Stream.of(this.quoteMapper.convert(quote))).collect(Collectors.toList());
	}

	public List<QuoteDto> getDailyQuotes(String symbol, LocalDate start, LocalDate end) {
		return this.dailyQuoteRepository.findBySymbolAndDayBetween(symbol, start, end).stream()
				.flatMap(quote -> Stream.of(this.quoteMapper.convert(quote))).collect(Collectors.toList());
	}

	public List<QuoteDto> getIntraDayQuotes(String symbol) {
		return this.intraDayQuoteRepository.findBySymbol(symbol).stream()
				.flatMap(quote -> Stream.of(this.quoteMapper.convert(quote))).collect(Collectors.toList());
	}
}
